samples/client/petstore/csharp-refactor/OpenAPIClient/src/Org.OpenAPITools/Client/ClientUtils.cs

```diff
@@ -192,6 +192,61 @@ public static byte[] ReadAsBytes(Stream inputStream)
         public static dynamic ConvertType(dynamic fromObject, Type toObject)
         {
             return Convert.ChangeType(fromObject, toObject);
-}
+        }
+
+        /// <summary>
+        /// Select the Content-Type header's value from the given content-type array:
+        /// if JSON type exists in the given array, use it;
+        /// otherwise use the first one defined in 'consumes'
+        /// </summary>
+        /// <param name="contentTypes">The Content-Type array to select from.</param>
+        /// <returns>The Content-Type header to use.</returns>
+        public static String SelectHeaderContentType(String[] contentTypes)
+        {
+            if (contentTypes.Length == 0)
+                return "application/json";
+
+            foreach (var contentType in contentTypes)
+            {
+                if (IsJsonMime(contentType.ToLower()))
```

Comment:
I think this might be incorrect… this would return true for `APPLICATION/JSON; CHARSET=UTF-8`, then apply that as a header when it's not going to be a useful or relevant header. Although HTTP spec defines header names (e.g. 'Content-Type') as case insensitive, the values are case sensitive.